### DISCUSSION TITLE
what do you think about https://github.com/reactive-python/reactpy?

### DISCUSSION THREAD

**Discussion by retsyo** [rtp-book/code#8] [Q&A] [answered]
Not to be offensive. I know very little about web development.

[reactpy](https://github.com/reactive-python/reactpy) says "It's React, but in Python", and is some active in development. Can anyone compare [reactpy](https://github.com/reactive-python/reactpy) with [REACT to PYTHON](https://pyreact.com/)? 

Thanks

**Comment by JennaSys** [comment, marked as answer]
Sure, ReactPy is a custom Python library that provides a framework for creating web applications. Behind the scenes, what it sends to the web browser is based on the JS React library. You don't really need to know anything about the JavaScript ecosystem, but you may be somewhat limited to what the framework provides.

The approach taken in React to Python utilizes React directly. There is no framework or wrapper around React. But, you are able to write all of your application code in Python instead of JavaScript by using the Transcrypt transpiler that converts your Python code into JavaScript during a build step. This approach allows you to use just about any JavaScript library you want. The downside is that you do need to know a little bit about the JavaScript ecosystem and understand the API of any JS libraries that you use.

In short, the Transcrypt approach from the book gives you more flexibility in how and what you use, but it requires a bit more knowledge about the underlying technologies. ReactPy hides a lot of that complexity from you and you just have to learn the ReactPy framework, but you are a bit more limited and need to follow the rules of the framework. Both are good approaches. It just depends on your needs and goals as to which one would work best for you.
